pywinauto/timings.py

```diff
@@ -389,3 +423,43 @@ def WaitUntilPasses(
     
     # return the function value
     return func_val
+	
+#=========================================================================
+def wait_until_passes(
+    timeout, 
+    retry_interval, 
+    exceptions = (Exception)):
+    
+    """
+	WaitUntilPasses decorator 
+    """
+    
+    def wait_until_passes_decorator(func):		
+        def wrapped(*args):
+            start = time.time()
+            # keep trying until the timeout is passed
+            while True:
+                try:
+                    # Call the function with any arguments
+                    func_val = func(*args)            
+                    # if this did not raise an exception -then we are finised
+                    break        
+                # An exception was raised - so wait and try again
+                except exceptions as e:        
+                    # find out how much of the time is left
+                    time_left = timeout - ( time.time() - start)        
+                    # if we have to wait some more        
+                    if time_left > 0:
+                        # wait either the retry_interval or else the amount of
+                        # time until the timeout expires (whichever is less)
+                        time.sleep(min(retry_interval, time_left))
+                    else:
+                        # Raise a TimeoutError - and put the original exception
+                        # inside it
+                        err = TimeoutError()
+                        err.original_exception = e
+                        raise err    
+            # return the function value
+            return func_val		
+        return wrapped
+    return wait_until_passes_decorator

```

Comment:
Could you look at the example of unit test for decorator. It is acceptable? Where I can place it?

    def test_wait_until():
        try:
            @wait_until(20, 10)
            def custom_foo():
                pass
            custom_foo()
        except TimeoutError as exception:
            return True
        return False